machine_usage/views.py

```diff
@@ -288,16 +289,32 @@ def generate_failed_usage_form(request):
                 output[type_name] = []
             output[type_name].append(m.machine_name)
 
+        # for m in machines_idle:
+        #     type_name = m.machine_type.machine_type_name
+        #     if type_name not in output:
+        #         output[type_name] = []
+        #     output[type_name].append(m.machine_name)
+
         return render(request, 'machine_usage/forms/failed_usage.html', {"machines_in_use":output})
     else:
         machine_name = request.POST["machine_name"]
         machine = Machine.objects.get(machine_name=machine_name)
 
-        usage = machine.current_job
-        usage.clear_time = timezone.now()
-        usage.failed = True
-        usage.status_message = "Failed."
-        usage.save()
+        fail_log = {
+            "machine": machine_name,
+            "user": request.user.get_username(),
+            "filament": request.POST["filament_type"],
```

Comment:
Couldn't you use `usage.slotusage_set.all()[0].resource` for filament or do you want the filament at time of failure in case that's the issue.

Reply:
filament at time of failure